File: solutions/066.plus-one/plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int length = digits.length, remain = 1;
        for(int i = length-1; i>=0; i--){
            remain = (digits[i]+1)/10;
            digits[i]=(digits[i]+1)%10;
            if(remain==0){
                break;
            }
        }
        
        if(remain>0){
            int[] res = new int[length+1];
            res[0] = remain;
            for(int i = 1; i< length+1;i++){
                res[i] = digits[i-1];
            }
            return res;
        }
        
        return digits;
    }
}

File: solutions/066.plus-one/plus-one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    let carry = 1;
    let index = digits.length - 1;
    while(carry !== 0 && index > -1){
        const temp = digits[index] + carry;
        digits[index] = temp % 10;
        carry = Math.floor(temp / 10);
        index--;
    }
    if(carry !== 0){
        digits.unshift(carry);
    }
    return digits;
};